File: local_examples/cmds_servermgmt/ioredis/cmds-servermgmt.js
// EXAMPLE: cmds_servermgmt

// HIDE_START
import assert from 'node:assert';
import { Redis } from 'ioredis';

const redis = new Redis();
// HIDE_END

// STEP_START flushall
const res1 = await redis.flushall(); // or flushall('ASYNC')
console.log(res1); // >>> OK

const res2 = await redis.keys('*');
console.log(res2); // >>> []
// STEP_END

// REMOVE_START
assert.equal(res1, 'OK');
assert.deepEqual(res2, []);
// REMOVE_END

// STEP_START info
const res3 = await redis.info();
console.log(res3);
// >>> # Server
// >>> redis_version:7.4.0
// >>> ...
// STEP_END

// REMOVE_START
assert.ok(res3.includes('redis_version'));
// REMOVE_END

// HIDE_START
redis.disconnect();
// HIDE_END

File: local_examples/cmds_servermgmt/ruby/cmds_servermgmt.rb
# EXAMPLE: cmds_servermgmt
# HIDE_START
require 'redis'

r = Redis.new
# HIDE_END

# REMOVE_START
def assert_equal(expected, actual)
  raise "Expected #{expected.inspect}, got #{actual.inspect}" unless actual == expected
end
# REMOVE_END

# STEP_START flushall
res1 = r.flushall # or r.flushall(async: true)
puts res1 # >>> OK

res2 = r.keys('*')
puts res2.inspect # >>> []
# STEP_END

# REMOVE_START
assert_equal('OK', res1)
assert_equal([], res2)
# REMOVE_END

# STEP_START info
res3 = r.info
puts res3['redis_version']
# >>> 7.4.0
# STEP_END

# REMOVE_START
raise 'info missing redis_version' unless res3.key?('redis_version')
r.close
# REMOVE_END
